### ISSUE TITLE
Update version for smartstring crate

### ISSUE THREAD

**Issue by nkkarpov** [sstadick/perbase#74]
The current version depends on undefined behavior (related to the string layout). This was fixed in SmartString v1.0. See the changes at https://github.com/bodil/smartstring/blob/master/CHANGELOG.md. The code that reproduces the issue in version 0.2.10:

```
use smartstring::{LazyCompact, SmartString};
fn main() {
    let a = vec![
        String::from("59281caf-dd63-4ec6-8123-87c07dff2089"),
        String::from("59281caf-dd63-4ec6-8123-87c07dff20890")];
    for v in a {
        println!("{:?}", v);
        println!("{:?}", SmartString::<LazyCompact>::from(v));
    }
}
/*
"59281caf-dd63-4ec6-8123-87c07dff2089"
"59281caf-dd63-4ec6-8123-87c07dff2089"
"59281caf-dd63-4ec6-8123-87c07dff20890"
"\0\0\0\0\0\0\0\0Ep/\u{1}\0\0\0%\0\0"
 */
```

**Comment by MatthiasZepper** [comment]
I would like to second the request to upgrade the SmartString dependency. When running perbase 0.10.0, I get occasional issues with the current version:
```

[2024-12-02T16:25:36Z INFO  perbase_lib::par_granges] Processing TID 5369:0-1861
[2024-12-02T16:25:36Z INFO  perbase_lib::par_granges] Processing TID 5370:0-1860
[2024-12-02T16:25:36Z INFO  perbase_lib::par_granges] Processing TID 5371:0-1858
[2024-12-02T16:25:36Z INFO  perbase_lib::par_granges] Processing TID 5372:0-1855
thread 'main' panicked at /opt/conda/conda-bld/perbase_1722270437824/work/.cargo/registry/src/index.crates.io-6f17d22bba15001f/smartstring-0.2.10/src/inline.rs:41:9:
assertion failed: len <= MAX_INLINE
stack backtrace:
   0:     0x55ef96f3201f - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h1e1a1972118942ad
   1:     0x55ef96e5823b - core::fmt::write::hc090a2ffd6b28c4a
   2:     0x55ef96f041b2 - std::io::Write::write_fmt::h8898bac6ff039a23
   3:     0x55ef96f33b09 - std::sys_common::backtrace::print::ha96650907276675e
   4:     0x55ef96f3332e - std::panicking::default_hook::{{closure}}::h215c2a0a8346e0e0
   5:     0x55ef96f34554 - std::panicking::rust_panic_with_hook::hac8bdceee1e4fe2c
   6:     0x55ef96f33e3a - std::panicking::begin_panic_handler::{{closure}}::h00d785e82757ce3c
   7:     0x55ef96f33dc9 - std::sys_common::backtrace::__rust_end_short_backtrace::h1628d957bcd06996
   8:     0x55ef96f33db6 - rust_begin_unwind
   9:     0x55ef96d677a2 - core::panicking::panic_fmt::hdc63834ffaaefae5
  10:     0x55ef96d6787b - core::panicking::panic::h75b3c9209f97d725
  11:     0x55ef96dc704f - <&mut csv::serializer::SeRecord<W> as serde::ser::SerializeStruct>::serialize_field::h0a12cac96608b6c1
  12:     0x55ef96dd1d65 - perbase::Subcommand::run::h19da3f0473014c9e
  13:     0x55ef96ddc0b4 - perbase::main::h883af21244aa9c0b
  14:     0x55ef96d975c3 - std::sys_common::backtrace::__rust_begin_short_backtrace::h0200b9b3b5d809f9
  15:     0x55ef96deafbf - main
  16:     0x14cc3d13f24a - <unknown>
  17:     0x14cc3d13f305 - __libc_start_main
  18:     0x55ef96d73ea6 - <unknown>
  19:                0x0 - <unknown>
```

However, when [building perbase from my own fork with upgraded dependencies](https://github.com/MatthiasZepper/perbase/tree/build_docker_own), the processing the same files works like a charm. 

Thanks for the great tool by the way!

**Comment by sstadick** [comment]
Thanks for the bump on this! I've merged the PR from @nkkarpov and created v0.10.1
